Core/src/org/sleuthkit/autopsy/coreutils/AppDBParserHelper.java

```diff
@@ -0,0 +1,1145 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.coreutils;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import org.apache.commons.lang3.StringUtils;
+import org.sleuthkit.autopsy.casemodule.Case;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.AccountFileInstance;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.Blackboard;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.Relationship;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskDataException;
+
+
+/**
+ * A helper class to support modules that parse SQLite databases from mobile 
+ * apps and create artifacts.
+ */
+public final class AppDBParserHelper  {
+    
+    private static final Logger logger = Logger.getLogger(AppDBParserHelper.class.getName());
+    
+    /**
+     * Enum for message read status
+     */
+    public enum MessageReadStatusEnum {
+
+        UNKNOWN,    /// read status is unknown
+        UNREAD,     /// message has not been read
+        READ        /// message has been read
+    }
+        
+    private final AbstractFile dbAbstractFile;
+    private final String moduleName;
+    
+    // 'self' account for the application. 
+    private final AccountFileInstance selfAccountInstance;
+    
+    // type of accounts to be created for the Application using this helper
+    private final Account.Type accountsType;
+    
+    /**
+     * Constructs a AppDB parser helper for the given DB file.
+     * 
+     * This is a constructor for Apps that that do not have any app specific account information 
+     * for device owner and will use  a 'Device' account in lieu.
+     * 
+     * It creates a DeviceAccount instance to use as a self account.
+     * 
+     * @param moduleName name module using the helper
+     * @param dbFile database file being parsed by the module
+     * @param accountsType account types created by this module
+     * 
+     * @throws TskCoreException 
+     */
+    public AppDBParserHelper(String moduleName, AbstractFile dbFile, Account.Type accountsType) throws TskCoreException {
+        
+        this.moduleName = moduleName;
+        this.dbAbstractFile = dbFile;
+        this.accountsType = accountsType;
+        this.selfAccountInstance = Case.getCurrentCase().getSleuthkitCase().getCommunicationsManager().createAccountFileInstance(Account.Type.DEVICE, ((DataSource)dbFile.getDataSource()).getDeviceId(), moduleName, dbFile);
```

Comment:
What happens if the user runs into a TskCoreException here? Is the idea that they skip it or would it be better if they followed up with a call to "public AppDBParserHelper(String moduleName, AbstractFile dbFile)", where this value is just null? If its the latter how would they know?